### ISSUE TITLE
OPENCV 4.0.0-beta onnx_importer.cpp:56: error: (-210:Unsupported format or combination of formats) Failed to parse onnx model in function 'ONNXImporter'

### ISSUE THREAD

**Issue by Con-Mi** [opencv/opencv#12869]
I try to export a PyTorch model from torchvision pretrained models into .onnx format. 
Then I try to read the network with cv::dnn::readNetFromONNX and I get the following error:

onnx_importer.cpp:56: error: (-210:Unsupported format or combination of formats) Failed to parse onnx model in function 'ONNXImporter'

Aborted (core dumped)

The Python code is:
```python
from torchvision import models
model = models.vggnet19_bn(pretrained=True)
dummy_input = torch.randn(1, 3, 224, 224)
torch.onnx.export(model, dummy_input, "vgg.onnx")
```

In C++:
```cpp
cv::dnn::Net = cv::dnn::readNetFromONNX("vgg.onnx");
```

Any ideas?
I have Opencv 4.0.0 beta and pytorch 4.0.1 cpu version

**Comment by dkurt** [comment]
@Con-Mi, Does `vgg.onnx` exists?

https://github.com/opencv/opencv/blob/2180a67670f50b6337e8f3c82f1ffb27b982110e/modules/dnn/src/onnx/onnx_importer.cpp#L51-L57

**Comment by dkurt** [comment]
@Con-Mi, You can make it faster providing a reference to `vgg.onnx` (in example, put it to a public share).

**Comment by Con-Mi** [comment]
@dkurt What do you mean? Share the binary vgg.onnx?

Also do I need to specify to build opencv with protobuf flag for using readNetFromONNX?

**Comment by dkurt** [comment]
@Con-Mi, just a binary `vgg.onnx` because I have no pytorch right now to generate it.

**Comment by Con-Mi** [comment]
@dkurt Check if this link works for you 
https://www.dropbox.com/s/oefl3kratmcf5wh/vgg.onnx?dl=0

**Comment by Con-Mi** [comment]
Anything came up yet? Any fixes?

**Comment by dkurt** [comment]
@Con-Mi, There are some nodes in this graph which are not mapped easily to deep learning layer. In example, `Shape` which takes a shape of input blob and then subsequent layers work with it but not with data. So we need more time to enable this model.

**Comment by charlesrwest** [comment]
I'm having a similar issue with my PyTorch generated model.  

Link for ONNX:
https://drive.google.com/file/d/1VPA3I1j4VZScp0lHh75EtF03wfy2HGmw/view?usp=sharing

Visualization of network:
https://drive.google.com/file/d/1zJ39RUuV0FOy-nxO90-UHklJmILwkq6P/view?usp=sharing

**Comment by charlesrwest** [comment]
I've stripped out the final FC layer of the resnet model I am trying to get to work and replaced it with a 1x1 conv layer.  

As far as I can tell, the only operators it uses are:
Conv
BatchNormalization
Relu
MaxPool
Add
AveragePool

So there isn't any reshape/slice operators.  However, it still fails with the same error.

Link for ONNX:
https://drive.google.com/file/d/1CkDQrk-dGlqtVC5zx7xDSLP3EP6RkMXP/view?usp=sharing

Visualization of network:
https://drive.google.com/file/d/1Mw68Dh83RyVtrFlqxT04PZQpNiomGW0t/view?usp=sharing

**Comment by dkurt** [comment]
@charlesrwest, Thank you! Could you please test the model with changes from https://github.com/opencv/opencv/pull/13002?

**Comment by charlesrwest** [comment]
This has been merged with master, right?

I pulled the latest master and recompiled opencv.  I still get the same error when I try to load the ONNX network:
terminate called after throwing an instance of 'SOMException'
  what():  Error of type UNKNOWN occurred in file /home/pi/cpp/projects/opencv/ONNXInference/src/executables/testONNXInference/main.cpp at line 50: Error, unable to load from ONNX fileOpenCV(4.0.0-pre) /home/pi/cpp/libraries/opencv/modules/core/src/matrix.cpp:321: error: (-215:Assertion failed) 0 <= d && d <= CV_MAX_DIM && _sizes in function 'create'

Aborted

**Comment by charlesrwest** [comment]
I just recompiled with the 3.4 branch and tried again.  It gave the same error.  

I also grepped the opencv source on my computer and found the patch to be in the version I compiled:
modules/dnn/src/onnx/onnx_importer.cpp:                // Paddings are in order begin0, begin1, .. beginN, end0, end1, ..., endN.

**Comment by Con-Mi** [comment]
I have build Opencv 3.4.3 from source and run the following script to generate the binary in .onnx format from a PyTorch model.

Python: 

from torchvision import models
import torch 

dense_model = models.densenet121(pretrained=True)
dummy_input = torch.randn(1, 3, 224, 224)

torch.onnx.export(alex_model, dummy_input, "dense.onnx")

I try to use the readNetFromONNX from OpenCV with the following:

#include <opencv2/dnn/dnn.hpp>
#include <opencv2/imgproc.hpp>
#include <opencv2/highgui.hpp>

int main() {
	const cv::String s = "../dense.onnx";
    cv::dnn::Net net = cv::dnn::readNetFromONNX(s);
    return 0;
}

I get the following error:

terminate called after throwing an instance of 'cv::Exception'
  what():  OpenCV(3.4.3-dev) opencv/modules/dnn/src/onnx/onnx_importer.cpp:271: error: (-204:Requested object was not found) Blob 1159 not found in const blobs in function 'getBlob'

Aborted (core dumped)

The link to the binary: 
https://www.dropbox.com/s/9exm74kdoqfun1m/dense.onnx?dl=0

**Comment by Con-Mi** [comment]
I have run the same for as above with OpenCV - 4.0.0-rc and I get the same error.
Is it a bug?

**Comment by dkurt** [comment]
@Con-Mi, this is a list of layers:
```
name type
...
1149 BatchNormalization
1150 Relu
1151 Conv
1152 Concat
1153 BatchNormalization
1154 Relu
1155 Pad
1156 AveragePool
1157 Constant
1158 Shape
1159 Gather
1160 Constant
1161 Unsqueeze
```
The problem is in the last set of layers:
```
1158 Shape
1159 Gather
1160 Constant
1161 Unsqueeze
```
